### PR TITLE
Demo website

### PR DESCRIPTION
**Pull request by jonthomp**
Adds the demo website homepage and fixes a few props on other components

Also published the demo site to github pages https://tabler.github.io/tabler-react/

screenshot from 2018-04-22 22-19-48

### CODE REVIEW

**Comment by AaronCoplan** [comment]
Our menu bar (Home interface components etc) looks and behaves differently than that of tabler/tabler.  Is this something we should try to resolve within this PR?

File: example/src/App.react.js [hunk 1]
```diff
@@ -25,7 +26,9 @@ type Props = {||};
 function App(props: Props): React.Node {
   return (
     <Router basename={process.env.PUBLIC_URL}>
-      <React.Fragment>
+      <Switch>
```
Comment (AaronCoplan) [inline example/src/App.react.js]: Good call adding switch 👍

File: example/src/HomePage.react.js [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+  Text,
+  Header,
+  Table,
+  Alert,
+  Progress,
+} from "tabler-react";
+
+import SiteWrapper from "./SiteWrapper.react";
+
+function Home() {
```
Comment (AaronCoplan) [inline example/src/HomePage.react.js]: add return type

**Comment by jonthomp** [comment]
@AaronCoplan that's a yes on the menu bar, the Tabler demo site has changed theirs since I built that bit, should have it done today

**Comment by AaronCoplan** [comment]
@jonthomp Ok, will you do that on this PR or a separate one?

**Comment by jonthomp** [comment]
@AaronCoplan I'll do it on this PR if that's ok?

**Comment by AaronCoplan** [comment]
@jonthomp Just let me know when this is ready to re-review!  And if you could update the screenshot that would be good

**Comment by jonthomp** [comment]
@AaronCoplan I think they're ready for review now, will update the screenshot.

During this update I started to run into issues with exact flow props and not spreading rest props so I've been trying to rethink/reduce how/when the RootComponent idea is used.

I've to come to trying to replace it with a new 'type' prop that limits your selection, for example`+type?: 'div' | 'li'`. Then, in this example, I've used this type to return either a real `<div>` or `<a>` tag rather than `RootComponent`.
However, for now at least, we do still need to use RootComponent in some instances but at the moment this is just to support react-router. Perhaps in the future, we remove it altogether and just provide separate react router specific components.

**Pull request merged**